src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncNodesRequestHandler.scala

```diff
@@ -77,7 +77,7 @@ class FastSyncNodesRequestHandler(
       val evm = n.getAccount.codeHash
       val storage = n.getAccount.storageRoot
 
-      mptNodeStorage.put(n)
+      nodesKeyValueStorage.update(Nil, Seq(n.hash -> n.toBytes))
```

Comment:
The storage trait is defined as dependency from the `MerklePatriciaTrie` and the clases to be stored here are from Network. 

I do agree it seems a little bit low level but the reason behind this is to remove the `MptNodeStorage` so I think the trade off was positive. If you want to, I can give you more background.